src/hotspot/share/oops/cpCache.cpp

```diff
@@ -623,32 +623,53 @@ Method* ConstantPoolCacheEntry::get_interesting_method_entry() {
   // print separator
   if (index == 0) st->print_cr("                 -------------");
-  // print entry
-  st->print("%3d  (" PTR_FORMAT ")  ", index, (intptr_t)this);
-  st->print_cr("[%02x|%02x|%5d]", bytecode_2(), bytecode_1(),
-               constant_pool_index());
-  st->print_cr("                 [   " PTR_FORMAT "]", (intptr_t)_f1);
-  st->print_cr("                 [   " PTR_FORMAT "]", (intptr_t)_f2);
-  st->print_cr("                 [   " PTR_FORMAT "]", (intptr_t)_flags);
-
-  if ((bytecode_1() == Bytecodes::_invokehandle ||
-       bytecode_1() == Bytecodes::_invokedynamic)) {
+  // print universal entry info
+  st->print_cr("%3d", index);
+  st->print_cr(" - this: " PTR_FORMAT, p2i(this));
+  st->print_cr(" - bytecode 1: %s %02x", Bytecodes::name(bytecode_1()), bytecode_1());
+  st->print_cr(" - bytecode 2: %s %02x", Bytecodes::name(bytecode_2()), bytecode_2());
+  st->print_cr(" - cp index: %5d", constant_pool_index());
+  if (is_method_entry()) {
     constantPoolHandle cph(Thread::current(), cache->constant_pool());
     Method* m = method_if_resolved(cph);
-    oop appendix = appendix_if_resolved(cph);
-    ResourceMark rm;
-    if (m != NULL) {
-      st->print_cr("  Method%s: " INTPTR_FORMAT " %s.%s%s",
-                   m->is_native() ? " (native)" : "",
-                   p2i(m),
-                   m->method_holder()->name()->as_C_string(),
-                   m->name()->as_C_string(), m->signature()->as_C_string());
-    }
-    if (appendix != NULL) {
-      st->print("  appendix: ");
-      appendix->print_on(st);
+    st->print_cr(" - F1:  [   " PTR_FORMAT "]", (intptr_t)_f1);
+    st->print_cr(" - F2:  [   " PTR_FORMAT "]", (intptr_t)_f2);
+    st->print_cr(" -     Method: " INTPTR_FORMAT " %s", p2i(m), m->external_name());
+    st->print_cr(" - flag values: [%02x|0|0|%01x|%01x|%01x|%01x|0|%01x|%01x|00|00|%02x]",
+               flag_state(), has_local_signature(), has_appendix(),
+               is_forced_virtual(), is_final(), is_vfinal(),
+               indy_resolution_failed(), parameter_size());
+    st->print_cr(" - tos: %s\n - local signature: %01x\n"
+          " - has appendix: %01x\n - forced virtual: %01x\n"
+          " - final: %01x\n - virtual Final: %01x\n - resolution Failed: %01x\n"
+          " - num Parameters: %02x",
+               type2name(as_BasicType(flag_state())), has_local_signature(), has_appendix(),
+               is_forced_virtual(), is_final(), is_vfinal(),
+               indy_resolution_failed(), parameter_size());
+    if ((bytecode_1() == Bytecodes::_invokehandle ||
+       bytecode_1() == Bytecodes::_invokedynamic)) {
+      oop appendix = appendix_if_resolved(cph);
+      if (m != NULL) {
+        st->print_cr("  Method%s: " INTPTR_FORMAT " %s.%s%s",
+                    m->is_native() ? " (native)" : "",
+                    p2i(m),
+                    m->method_holder()->name()->as_C_string(),
+                    m->name()->as_C_string(), m->signature()->as_C_string());
```

Comment:
You removed the ResourceMark needed by `as_C_string` so they won't get cleaned up at this level (and will cause a failure if there isn't a RM somewhere in the call stack).